### ISSUE TITLE
No events are recorded if filter is not set at the first execution

### ISSUE THREAD

**Issue by kumaraswamygaviyappa** [benhar-dev/tchmi-export-events#1]
Hello,

I have observed at the beginning that, 

- **Export to csv** button pressed, resulted in empty csv with only header.
- Added the filter, Events are recorded in csv. eg. if filter is ADS, all ADS events are recorded.
- Deleted previous ADS filter, all events are recorded in csv.  

Observed same behaviour on another new project which created to test this issue.

Am i making any mistake?

**Comment by benhar-dev** [comment]
Hi @kumaraswamygaviyappa, thank you for mentioning this.  

The first time it is run we are passing null in to the function.  The code below will see that the argument Filter is null and as such will use defaultFilter in it's place.  How this works can be found [here](https://javascript.info/nullish-coalescing-operator#comparison-with).
```javascript
 request.commands[0].filter = Filter || defaultFilter;
```
The default filter is as follows.
```javascript
var defaultFilter = [
    { "path": "type", "comparator": "!=", "value": 2 },
    { "logic": "AND" },
    [
        {
            "path": "domain",
            "comparator": "==",
            "value": "TcHmiEventLogger"
        }
    ]
]
```

When you remove the filter (on your 3rd step) you are not actually fully removing it as the filter dialog box will give you an empty array as an output.  As such an empty array is not null, which means that the empty array will be used as the filter, which means you now see everything.  

So, to summarize, what you are seeing is correct.  If in your code you would prefer the default filter to show everything then you will need to modify it to read this.
```javascript
var defaultFilter = []
```

The reason behind all of this is because if you pass null in to the server request then it will fail, so I added code to make sure that something was provided.  In hindsight I should have made the defaultFilter = [] as it would have been less confusing, but I wanted to also show you what the filter looks like in code.  

I hope this helps.

**Comment by kumaraswamygaviyappa** [comment]
Hello @benhar-dev ,

That's amazing and thank you for the detailed explanation.
I will change `var defaultFilter = []`.
